docs/DeveloperGuide.md

```diff
@@ -5,9 +5,238 @@
+
+The Class diagram for Hydration list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![HydrationListClassDiagram.png](HydrationListClassDiagram.png)
+
+### Hydration delete feature
+
+The `hydration delete` feature can delete the hydration record at specific index of hydration list. This functionality is facilitated by `HydrationList`. It implements one operation, namely:
+- `deleteEntry(String line)`
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `hydration delete INDEX` in the terminal, the string is sent to `Ui#handleUserInput()`, which will call `Ui#handleHydrationInput()`.
+
+- Step 2: After the `Ui#handleHydrationInput()` matching `delete hydration` key word, the string will be passed into deleteEntry(String line) to execute delete process.
+
+- Step 3: The string will be divided to two substrings according to the command syntax. Index will be tried to get from the second substring by `Integer.parseInt()`.
+
+- Step 4: The hydration record (`Entry`) stored in the `ArrayList<Entry> hydrationList` will be deleted by calling `hydrationArrayList.remove((index-1));` and a successful deleting message will be shown in terminal by calling `HydrationListUi#successfulDeletedMessage(toDelete)`
+
+- Step 5: The latest hydration list will be updated to saving file by calling `HydrationList#updateFile()`.
+
+The Sequence diagram for Hydration delete feature is shown below:
+![HydrationDeleteDiagram.png](HydrationDeleteDiagram.png)
+
+### Adding sleep entries feature
+
+#### Implementation
+
+This functionality is facilitated by `UI`, `SleepList`, `FileHandler` and `ParserSleep`. It implements one operation, namely:
+- `UI#handleSleepInput(String,SleepList)`
+- `SleepList#addEntry(String)`
+- `ParserSleep#parseSleepInput(String)`
+- `FileHandler#updateFile()`
+
+This feature is activated when the user inputs a `sleep add` command in the terminal.
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+
+- Step 1: When the user inputs the command `sleep add 7.5 d/2022-01-02` in the terminal,
+  the string is sent to `UI#handleSleepInput(String, SleepList)`, which calls `SleepList#addSleep(String)`.
+
+- Step 2: Inside `SleepList#addSleep(String)`, the function `ParserSleep#parseSleepInput(String)` is then called to extract information such as the duration and date of entry.
+
+- Step 3: It will create a new entry of class `SleepEntry` that extends `Entry`based on the information.
+
+- Step 4: The created `SleepEntry` instance is added into the `ArrayList<Entry> sleepList` attribute of the `SleepList`.
+
+- Step 5: `FileHandler#updateFile()` is then called to update the data file with the new entry in the `SleepList`.
+
+The sequence diagram for this feature is shown below:
+![SleepAddSeqDiagram](https://github.com/a-wild-chocolate/tp/blob/f9a94746e763ddf54a4309583b71e0fdbabdb141/docs/SleepAddSeqDiagram.jpg)
```

Comment:
Standardize the tools used for the diagrams.
A few issues: 

1. No return from activation bar

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/69704055/0f86b1f9-9d44-474f-8851-3f178690aca6)

2. Return line not starting at end of activation bar. 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/69704055/18ff9674-238e-45c9-b355-8236e9d0023d)

3. User does not need an activation bar